商品削除機能

KDItsu opened the pull request:
# What
商品削除機能画面の作成

# Why
商品削除機能を実装するため

■ログイン状態の出品者のみ、詳細ページの削除ボタンを押すと、出品した商品を削除できる動画
https://gyazo.com/3bc2d67ac0ed83db2d1fd066eef7befb

KDItsu committed:
destroyアクションのルーティングを設定

resourcesは、7つのアクションへのルーティングを自動生成するメソッドです。

KDItsu committed:
削除ボタンを実装する

KDItsu committed:
destroyアクションを定義する

yuri240 commented on app/controllers/items_controller.rb:
```diff
@@ -35,6 +35,12 @@ def update
   
+  def destroy
+    item = Item.find(params[:id])
+    item.destroy
```
◎出品者かどうかを判別する条件式がdestroyアクションにも適応されているか確認しましょう。

■理由
「「ログイン状態の場合にのみ、自身が出品した商品情報を削除できること」とは、詳細ページにおける「削除」ボタンの表示・非表示に加え、コントローラー側でも条件を設けることを指す」という実装条件を満たす為です。

KDItsu replied:
ありがとうございます。修正しました。

yuri240 reviewed (commented):
レビュー依頼ありがとうございます！
コメントさせていただきましたので、ご確認お願いいたします。

`※修正後はコミット、プッシュを忘れずに行い、改めてレビュー依頼フォームの送信をお願い致します※`

KDItsu committed:
出品者かどうかを判別する条件式

◎出品者かどうかを判別する条件式がdestroyアクションにも適応されているか確認しましょう。

■理由
「「ログイン状態の場合にのみ、自身が出品した商品情報を削除できること」とは、詳細ページにおける「削除」ボタンの表示・非表示に加え、コントローラー側でも条件を設けることを指す」という実装条件を満たす為です。

CHURO-MAU commented:
レビュー依頼ありがとうございます！

LGTMとさせていただきます！
削除機能の実装、お疲れ様でした！

マージできましたら、ローカルのmasterブランチへ、忘れずにプルしましょう。
そのあと、新しいブランチを作成し、次の実装に取り組んでください。

いよいよ最後の機能ですね！
がんばってください！
